pkg/ddl/executor.go

```diff
@@ -4331,17 +4329,17 @@ func (e *executor) renameTables(ctx sessionctx.Context, oldIdents, newIdents []a
 	}
 
 	job := &model.Job{
+		Version:             model.GetJobVerInUse(),
 		SchemaID:            schemas[1].ID,
-		TableID:             tableIDs[0],
+		TableID:             infos[0].TableID,
 		SchemaName:          schemas[1].Name.L,
 		Type:                model.ActionRenameTables,
 		BinlogInfo:          &model.HistoryInfo{},
 		CDCWriteSource:      ctx.GetSessionVars().CDCWriteSource,
-		Args:                []any{oldSchemaIDs, newSchemaIDs, tableNames, tableIDs, oldSchemaNames, oldTableNames},
-		CtxVars:             []any{append(oldSchemaIDs, newSchemaIDs...), tableIDs},
 		InvolvingSchemaInfo: involveSchemaInfo,
 		SQLMode:             ctx.GetSessionVars().SQLMode,
 	}
+	job.FillArgs(&model.RenameTablesArgs{RenameTableInfos: infos})
 
 	err = e.DoDDLJob(ctx, job)
```

Comment:
call e.doDDLJob2() and save RenameTableArgs into jobWraper.

Reply:
We cannot use `DoDDLJob2` here since `appendToSubJobs` still uses `job.Args`.  Maybe I can leave a TODO here.